Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1068,6 +1068,46 @@ LayoutUnit LineBuilder::adjustGeometryForInitialLetterIfNeeded(const Box& floatB
     return initialLetterCapHeightOffset.value_or(0_lu);
 }
 
+bool LineBuilder::placedFloatsContainsLeftPositionedFloat() const
+{
+    ASSERT(floatingState());
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    for (auto floatItem : m_placedFloats) {
+        if (floatingContext.toFloatItem(floatItem->layoutBox()).isLeftPositioned())
+            return true;
+    }
+    return false;
+}
+
+bool LineBuilder::placedFloatsContainsRightPositionedFloat() const
+{
+    ASSERT(floatingState());
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    for (auto floatItem : m_placedFloats) {
+        if (floatingContext.toFloatItem(floatItem->layoutBox()).isRightPositioned())
+            return true;
+    }
+    return false;
+
+}
```

Comment:
FloatingState already has hasLeftPositioned/hasRightPositioned functions. I think you could use them instead.